### ISSUE TITLE
Wildcard/ACME v2 support

### ISSUE THREAD

**Issue by mKeRix** [janeczku/rancher-letsencrypt#110]
Let's encrypt just launched the new API with wildcard support: https://community.letsencrypt.org/t/acme-v2-and-wildcard-certificate-support-is-live/55579. Certbot 0.22.0 and above already support the new features.

Would love to see these upstream in the rancher-letsencrypt project!

**Comment by willseward** [comment]
FWIW: https://github.com/xenolf/lego/issues/457

**Comment by blackholegalaxy** [comment]
@janeczku any news on that?

**Comment by Jefnull** [comment]
+1

**Comment by KryptionX** [comment]
If we could get this updated to support wildcards that would be extremely awesome.

**Comment by natiz** [comment]
+1

**Comment by nbejansen** [comment]
+1

**Comment by Chuckame** [comment]
Yes, wildcard support will awesome ! @janeczku , are u alive ?! ^^

**Comment by bearmoo-cloud-net** [comment]
+1

**Comment by B-Stefan** [comment]
xenolf/lego#457 just got merged.
Do we need to change here some code or just update the package?

**Comment by asdek** [comment]
If you need to generate wildcard certificates with using ACME v2 then you can using these repos:

- [git vxcontrol/rancher-letsencrypt](https://github.com/vxcontrol/rancher-letsencrypt)
- [docker vxcontrol/rancher-letsencrypt](https://hub.docker.com/r/vxcontrol/rancher-letsencrypt/)
- [release page v1.0.0](https://github.com/vxcontrol/rancher-letsencrypt/releases/tag/v1.0.0)

For using docker repo:
`docker pull vxcontrol/rancher-letsencrypt`

I think that making a Pull Request does not make sense ((

**Comment by Chuckame** [comment]
Why use a forked repo when we can have the official repo up-to-date with
wildcard support ? This is the purpose of a PR (I think)..

**Comment by blackholegalaxy** [comment]
@Chuckame because there is no vital sign from the author and repo maintainer to accept a PR?

**Comment by asdek** [comment]
@Chuckame I made [PR#114](https://github.com/janeczku/rancher-letsencrypt/pull/114) to main repo. Let's see together how quickly it will be merged.

@blackholegalaxy I fully support your opinion.

**Comment by jotolo** [comment]
@asdek, Is there any catalog including your repository with the changes? I think it's a good idea to create a catalog for that, like this repository has in rancher community catalog.
Thanks anyways for the update!

**Comment by asdek** [comment]
@jotolo, You can use `https://github.com/vxcontrol/rancher-catalog.git` repo and branch `public`. Or you can change field `Select Image` to `vxcontrol/rancher-letsencrypt:v1.0.0` after deploying package from community catalog.
PS I very much hope that the author will accept the changes and this will be available in the original package.

**Comment by jotolo** [comment]
@asdek I did the second option and it worked like charm! Thank you for the PR! I hope those who have the same issue can use this solution.It's very good to have it! I hope like you the PR to be merged.
Thanks again @asdek

**Comment by bearmoo-cloud-net** [comment]
@asdek 

I replace docker image to docker vxcontrol/rancher-letsencrypt. I ran the image and I'm getting this error:

6/12/2018 7:24:04 PMtime="2018-06-13T02:24:04Z" level=info msg="Using HTTP challenge: Sleeping for 120 seconds before requesting certificate"
6/12/2018 7:24:04 PMtime="2018-06-13T02:24:04Z" level=info msg="Make sure that HTTP requests for '/.well-known/acme-challenge' for all certificate domains are forwarded to port 80 of the container running this application"
6/12/2018 7:26:04 PMtime="2018-06-13T02:26:04Z" level=info msg="Trying to obtain SSL certificate (example.net,*.example.net) from Let's Encrypt Production CA"
6/12/2018 7:26:04 PMtime="2018-06-13T02:26:04Z" level=info msg="[INFO][example.net, *.example.net] acme: Obtaining bundled SAN certificate"
6/12/2018 7:26:05 PMtime="2018-06-13T02:26:05Z" level=info msg="[INFO][*.example.net] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/cLnZCIg3jc_l16zduayBdwa1tK1xEDB_haUNZohrN_c"
6/12/2018 7:26:05 PMtime="2018-06-13T02:26:05Z" level=info msg="[INFO][example.net] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/wy7n1Q5P34bpcdtkPx5TCeJM82C8dikP3uzbuW5ECwc"
6/12/2018 7:26:05 PMtime="2018-06-13T02:26:05Z" level=info msg="[INFO][example.net] acme: Authorization already valid; skipping challenge"
6/12/2018 7:26:05 PMtime="2018-06-13T02:26:05Z" level=info msg="[INFO][example.net] acme: Could not find solver for: dns-01"
6/12/2018 7:26:05 PMtime="2018-06-13T02:26:05Z" level=error msg="[acme: Error -> One or more domains had a problem:\n[example.net] [example.net] acme: Could not determine solvers\n] Error obtaining certificate: acme: Error -> One or more domains had a problem:\n[example.net] [example.net] acme: Could not determine solvers\n"

**Comment by asdek** [comment]
@bearmoo-repo Which one provider you use?

**Comment by bearmoo-cloud-net** [comment]
@asdek it would be no-ip

**Comment by asdek** [comment]
@bearmoo-repo I was asking about DNS provider. Because I tested on Route53 and CloudFlare, the problem was observed once with the fact that CloudFlare did not have time to update the DNS-record. I can try to incorporate some delay and define it through Environment Variable.

**Comment by bearmoo-cloud-net** [comment]
www.noip.com is my DNS provider.

**Comment by bearmoo-cloud-net** [comment]
Is it all possible to get the information that I need to add into TXT?

**Comment by asdek** [comment]
> www.noip.com is my DNS provider.

ok, thanks. I'll see what can do here.
If you want to add a TXT-record manually, you can use this service: https://www.sslforfree.com/

> cLnZCIg3jc_l16zduayBdwa1tK1xEDB_haUNZohrN_c
> wy7n1Q5P34bpcdtkPx5TCeJM82C8dikP3uzbuW5ECwc

In your log most likely these 2 values should be added within the TXT-records.

**Comment by solocommand** [comment]
@janeczku Would you be willing to make @asdek or another contributor a maintainer of this project to allow it to move forward?

**Comment by JoelESvensson** [comment]
The v1 API was just deactivated so renewals stopped working. Fortunately it seems like https://hub.docker.com/r/vxcontrol/rancher-letsencrypt works as a drop-in replacement. I recommend everyone having problems to upgrade to it instead. It being `vxcontrol/rancher-letsencrypt:v1.0.0`

**Comment by grabekm90** [comment]
I guess https://github.com/TrueCarry/rancher-letsencrypt/tree/acme-v2 is better updated

**Comment by bkuhl** [comment]
ACME v2 - For anyone still trying to do this, check out https://gist.github.com/fridgerator/db607d268f1f99329c8f9449e89abb4f
